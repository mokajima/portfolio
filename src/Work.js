import React from 'react'

const Work = (props) => {
  const {work} = props
  const className = work.border ? 'work__img work__img--border' : 'work__img'

  return (
    <li className="featured-work__item work">
      <a href={work.url} target="_blank">
        <figure className={className}>
          <img src={`images/${work.id}.jpg`} alt={work.title} />
        </figure>
      </a>
      <h3 className="work__title">
        <a href={work.url} target="_blank">{work.title}</a>
      </h3>
      <p className="work__summary">{work.summary}</p>
    </li>
  )
}

export default Work
