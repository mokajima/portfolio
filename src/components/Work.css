.work {
  text-align: center;
}

.work__img {
  border-radius: 10px;
  box-sizing: border-box;
  display: block;
  margin-bottom: 20px;
  overflow: hidden;
  width: 100%;
  position: relative;
  z-index: 1;
}

.work__img--border {
  border: 1px solid #ddd;
}

.work__img:hover img {
  transform: scale(1.1);
}

.work__img img {
  display: block;
  backface-visibility: hidden;
  width: 100%;
  height: auto;
  transition: transform 0.5s ease;
}

.work__title {
  font-size: 1.5em;
  margin-bottom: 20px;
}

.work__summary {
  line-height: 1.7;
}
