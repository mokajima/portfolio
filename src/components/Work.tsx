import React, { FC } from 'react'
import './Work.css'

interface Props {
  work: Work
}

const Work: FC<Props> = ({
  work
}) => {
  const className = work.border ? 'work__img work__img--border' : 'work__img'

  return (
    <div className="work">
      <a href={work.url} target="_blank" rel="noopener noreferrer">
        <figure className={className}>
          <img src={`images/${work.id}.jpg`} alt={work.title} />
        </figure>
      </a>
      <h3 className="work__title">
        <a href={work.url} target="_blank" rel="noopener noreferrer">{work.title}</a>
      </h3>
      <p className="work__summary">{work.summary}</p>
    </div>
  )
}

export default Work
