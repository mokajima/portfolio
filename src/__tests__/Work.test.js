import React from 'react'
import Work from '../components/Work'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const work = {
    id: 'readable',
    title: 'Readable',
    summary: 'An anonymous content and comment web application.',
    url: 'https://github.com/mokajima/readable',
    border: true,
    date: '2018-12'
  }

  let tree = renderer
    .create(<Work work={{ ...work, border: true }} />)
    .toJSON()
  expect(tree).toMatchSnapshot()

  tree = renderer
    .create(<Work work={{ ...work, border: false }} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
