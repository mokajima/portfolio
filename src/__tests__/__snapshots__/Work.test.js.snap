// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="work"
>
  <a
    href="https://github.com/mokajima/readable"
    rel="noopener noreferrer"
    target="_blank"
  >
    <figure
      className="work__img work__img--border"
    >
      <img
        alt="Readable"
        src="images/readable.jpg"
      />
    </figure>
  </a>
  <h3
    className="work__title"
  >
    <a
      href="https://github.com/mokajima/readable"
      rel="noopener noreferrer"
      target="_blank"
    >
      Readable
    </a>
  </h3>
  <p
    className="work__summary"
  >
    An anonymous content and comment web application.
  </p>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="work"
>
  <a
    href="https://github.com/mokajima/readable"
    rel="noopener noreferrer"
    target="_blank"
  >
    <figure
      className="work__img"
    >
      <img
        alt="Readable"
        src="images/readable.jpg"
      />
    </figure>
  </a>
  <h3
    className="work__title"
  >
    <a
      href="https://github.com/mokajima/readable"
      rel="noopener noreferrer"
      target="_blank"
    >
      Readable
    </a>
  </h3>
  <p
    className="work__summary"
  >
    An anonymous content and comment web application.
  </p>
</div>
`;
